automerge-frontend/src/lib.rs

```diff
@@ -155,68 +197,78 @@ impl FrontendState {
     {
         match self {
             FrontendState::WaitingForInFlightRequests {
-                mut in_flight_requests,
-                reconciled_root_state,
+                in_flight_requests,
+                reconciled_root_state: _,
                 optimistically_updated_root_state,
+                seen_non_local_patch: _,
                 max_op,
             } => {
-                let mut mutation_tracker = mutation::MutationTracker::new(
-                    optimistically_updated_root_state,
-                    max_op,
-                    actor.clone(),
-                );
+                let mut state_clone = optimistically_updated_root_state.clone();
```

Comment:
(Doesn't have to be fixed in this PR though)

Reply:
Ok, I'm looking at implementing rollback into the mutation tracker struct so we can recover this in case of failure, should also be cheaper (in the error case) than cloning the entire state again.